### ISSUE TITLE
@JsonProperty(required = false) is ignored

### ISSUE THREAD

**Issue by JoeryH** [mbknor/mbknor-jackson-jsonSchema#142]
I am using Kotlin, but I think this applies to all languages:

```kotlin
@JsonProperty(required = false)
val optionalDefaultWithAnnotation: String = "Hello"
```

The resulting shema will include this field as required, effectively ignoring the annotation.

**Comment by almson** [comment]
Possible duplicate of https://github.com/mbknor/mbknor-jackson-jsonSchema/issues/132

Not sure about Kotlin and `val`, but similar thing happens for non-boxed primitives. Basically this field is always present when you're serializing an object, but optional when deserializing. Unfortunately, there's no way to tell jsonSchema which kind of schema you're looking for.

**Comment by harrismirza** [comment]
Yeah, I've hit this with primitives as well, maybe it's worth having an config option for not forcing primitives to be required?

**Comment by harrismirza** [comment]
This might also be worth making a seperate issue, but if you set a default value for a primitive through the use of @JsonSchemaDefault, it also adds the property as required.
